packages/syft/src/syft/core/tensor/lazy_repeat_array.py

```diff
@@ -231,3 +373,18 @@ def transpose(self, *args: List[Any], **kwargs: Dict[str, Any]) -> lazyrepeatarr
         return lazyrepeatarray(
             data=self.data.transpose(*args, **kwargs), shape=dummy_res.shape
         )
+
+    def add_op(self, function: str, selection=slice(None), args=None) -> None:  # type: ignore
+        self.transforms.append((function, selection, args))
+
+    def evaluate(self):  # type: ignore
+        result = np.ones(self._shape) * self.data
+        for func, selection, args in self.transforms:
+            if func == "np.matmul":
+                if isinstance(args, lazyrepeatarray):
+                    args = args.to_numpy().evaluate()
+                result = eval(func)(result[selection], args)
```

Comment:
We should be really careful about using `eval` I think we should discuss another way.